### PR TITLE
Javascript: Project Todo

### PR DESCRIPTION
**Pull request by emmanuelkamala**
In this project:

A user can add a todo and inside can create a list of todos related to a todo.

- A user can edit a todo list
- A user can create a due date
- A user can set a priority of a todo
- A user can delete a todo

### CODE REVIEW

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED 

Hi @emmanuelkamala good job working on this, everything works fine :+1:  but there is just one thing left, Github actions is not set for this project so there is no check for errors,  please look into setting Github actions, do this and request for another review :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask:

**Comment by emmanuelkamala** [comment]
[@Oghenebrume50 ](https://github.com/Oghenebrume50) What do you mean by GitHub actions?, I have not seen that requirement from the project description.

**Comment by shloch** [comment]
I see that it's not yet emphasised on the requirements 
Read about it here --> [https://microverse.zendesk.com/hc/en-us/articles/360049654493](https://microverse.zendesk.com/hc/en-us/articles/360049654493) 

(just for the knowledge)

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @emmanuelkamala , 
Thanks for including Stickler.....am sorry, i really want to give a thumbs up, but i get a console error...... delete all your cache at your level and test project again,

Anyways i hope i am not alone..........This is what i get after trying several times
Please kindly debug it

Screen Shot 2020-06-10 at 11 16 24 PM

**Comment by emmanuelkamala** [comment]
Hi, 
I have corrected the error that was causing the app to not create projects. Now it is working fine.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @emmanuelkamala  awesome effort getting everything to work at this level, :clap:  I think you have done really well, there is just one thing left to do :slightly_smiling_face: 

The todo should be editable see this I currently see an edit icon but the implementation has not be made yet it still toggles :slightly_smiling_face: 

Please look into that and rrequest for another review :+1: 

PS: About the Github action that was my error, I felt it has been made mandatory already, so sorry :pray:  :slightly_smiling_face: 


**RAPHAEL NORIODE** :mask:  stay_safe

**Comment by emmanuelkamala** [comment]
I have corrected the 'Edit' Functionality in the app that was not working correctly.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @emmanuelkamala awesome work this looks good to me, but please fix the failing linter check, it is very important. 

Please make changes and request for another review :+1:  

**RAPHAEL NORIODE** :mask: stay_safe

**Comment by emmanuelkamala** [comment]
These ESLinting errors are not making sense to me, no matter what I try to correct what they say, it's not working and some of them are meaningless, like they say, 'import' keyword is a reserved keyword, or I should not use 'let' or 'const'. I have tried to configure a parser that converts it to the old ES5 syntax and makes it read like ES6 but, it still is not working. I have tried googling the solution but can't seem to find one that is working. Please help if you can understand what it means and how to correct it.

**Comment by emmanuelkamala** [comment]
All the errors that the ESLint is projecting, don't make sense totally. It says, I should have space between my if and the brackets, while none of my code is not spaced between the brackets, also it says, IIFE should have parenthesis, while all my IIFEs have that, it says, I should not use var, while I have not used var in any of my code. What should I correct then? and what is the point of this ESLint if itself is not making any sense?

**Comment by emmanuelkamala** [comment]
It's like ESLint is just wasting time trying to correct stuff that are just correct from the start.

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @emmanuelkamala,

**Please fix the linter errors as mentioned, otherwise, the project cannot be approved**
I'll reach out to you on slack so we can fix linter errors

## Status: Changes Required ♻️

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      "no-shadow": "off",
+      "no-param-reassign": "off",
+      "eol-last": "off",
+      "arrow-parens": "off"
+    },
+    "ignorePatterns": [
+      "dist/",
+      "build/",
+      "src/",
+      "webpack-config.js"
```
Comment (shshamim63) [inline .eslintrc.json]: Please do not make any changes to the config files. 
image
For resolving them please read the error messages carefully the solution is also in there.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## :recycle: Required Changes :recycle: 
___
Hi @emmanuelkamala :wave: 
Your project is complete based on the functionality :clap: . But based on the linter configuration, you can not change the configuration of it. If you think that any rules are necessary to include please create an issue in this [link](https://github.com/microverseinc/linters-config/tree/master/javascript). Please remove the changes you have added on the config file and resolve the errors. After that send the final code review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by emmanuelkamala** [comment]
I accidentally submitted a milestone that was one finished, as I had so many milestones to submit for completion, and I ended up submitting this one which was not done yet.

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @emmanuelkamala you have done really well with this, kudos for making changes for every review done on this project, this looks good to me now, please merge and move on

**This is approved**

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Comment by emmanuelkamala** [comment]
@Oghenebrume50, @shloch, @fatymahmed, @shshamim63 Thank you all.

**Pull request merged**